lib/themes/dosomething/paraneue_dosomething/js/components/ReactionsList.js

```diff
@@ -0,0 +1,158 @@
+const React = require('react');
+
+import Reaction from './Reaction';
+import ApiClient from '../utilities/ApiClient';
+import Setting from '../utilities/Setting';
+
+/**
+ * ReactionsList Component
+ * <ReactionsList />
+ */
+class ReactionsList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.apiClient = new ApiClient('v1');
+    this.approvedReactions = this.getApprovedReactions();
+    this.enabled = Setting('dsKudosReactions.enabled', false);
+
+    this.state = {
+      reactions: this.setInitialReactions()
+    };
+
+    // Bindings
+    this.renderReaction = this.renderReaction.bind(this);
+    this.toggleReaction = this.toggleReaction.bind(this);
+  }
+
+  /**
+   * Get all approved reactions from list of available reactions.
+   *
+   * @return {Object}
+   */
+  getApprovedReactions() {
+    const reactions = {};
+    const approvedReactions = Setting('dsKudosReactions.approved', []);
+    const availableReactions = Setting('dsKudosReactions.terms');
+
+    approvedReactions.forEach((value) => {
```

Comment:
Could this be simplified down like this? Although it would lose the custom keying, not sure if that's necessary.

```js
return approvedReactions.filter(value => availableReactions.hasOwnProperty(value));
```